test/cpp/ext/gcp/environment_autodetect_test.cc

```diff
@@ -124,17 +118,17 @@ TEST_F(EnvironmentAutoDetectTest, AppEngine) {
 TEST_F(EnvironmentAutoDetectTest, MultipleNotifyWaiters) {
   grpc::internal::EnvironmentAutoDetect env("project");
 
-  absl::Notification notify[10];
+  grpc_core::Notification notify[10];
   for (int i = 0; i < 10; ++i) {
     GetNotifiedOnEnvironmentDetection(&env, &notify[i]);
   }
   for (int i = 0; i < 10; ++i) {
     notify[i].WaitForNotification();
   }
 
-  EXPECT_EQ(env.resource()->resource_type, "global");
-  EXPECT_THAT(env.resource()->labels,
-              UnorderedElementsAre(Pair("project_id", "project")));
+  // Unless we test in a specific GCP resource, we should get "global" here.
+  // EXPECT_EQ(env.resource()->resource_type, "global");
```

Comment:
Same question here.

Reply:
This is a problem of our testing environments. I can't be sure whether we'd be running on an environment that has MetadataServer or not, so unless I mock the MetadataServer, I can not get a stable result here across environments.